test/unified_format.py

```diff
@@ -470,9 +470,21 @@ def __init__(self, test_class):
 
     def _operation_exists(self, spec, actual, key_to_compare):
         if spec is True:
-            self.test.assertIn(key_to_compare, actual)
+            if key_to_compare is None:
```

Comment:
Why would key_to_compare be None?

Reply:
It gets set to `None` [here](https://github.com/mongodb/mongo-python-driver/blob/e02eb287e896ff0e301e67648ba9398f6fe5d799/test/unified_format.py#L544) when the outer operation is special, as is the case with:

```
"$$unsetOrMatches": {
   "$$exists": true
}
```

Reply:
Hmm, the existing code here is pretty confusing to me. I would have expected key_to_compare to be the key $$unsetOrMatches is comparing against. Is this a bug in _operation_unsetOrMatches?

Reply:
Okay, by the time we get to this point we are already comparing the value of the key.  For example, if the key was `resumeAfter`, we're checking the value of the `resumeAfter`, which is `{'_data': '82628D384B000000022B0229296E04'}`, against the condition `{{ $exists: true }}`.  

This is for the `test.test_change_stream.TestUnifiedChangeStreamsResumeAllowlist.test_change_stream_resumes_after_CursorNotFound` test case.  The triggering spec is:

```json
"resumeAfter": {
  "$$unsetOrMatches": {
    "$$exists": true
  }
}
```

Reply:
Done and done